packages/calcite-components/src/components/sheet/sheet.tsx

```diff
@@ -245,6 +250,19 @@ export class Sheet
 
   // #region Private Methods
 
+  private keyDownHandler = (event: KeyboardEvent): void => {
+    const { defaultPrevented, key } = event;
+
+    if (defaultPrevented || this.escapeDisabled || !this.focusTrapDisabled) {
+      return;
+    }
+
+    if (this.open && key === "Escape") {
```

Comment:
Nitpick: you can move this condition in the `if` statement above to have a single `if` block. Applies to similar event handlers.